registrations/test_views.py

```diff
@@ -235,6 +235,99 @@ def test_get_registration_only_by_user(self):
+    def test_facility_code_check(self):
+
+        """
+            Test on Facility Code Check
+            GET - returns if facility code is correct, else return 200 response
+        """
+        self.make_source_normaluser()
+
+        clinic_code = 123456
+
+        result = {
+            "title": "Facility Code Check",
+            "headers": [],
+            "rows": [[clinic_code, "abcdefg", "test facility code"]],
+            "width": 1,
+            "height": 1,
+        }
+        responses.add(
+            responses.GET,
+            "http://jembi/ws/rest/v1/facilityCheck?{}".format(
+                urlencode({"criteria": "value:{}".format(clinic_code)})
+            ),
+            json=result,
+            status=200,
+        )
+        url = "{}?{}".format(
+            reverse("facilitycode-check"), urlencode({"clinic_code": clinic_code})
+        )
+        response = self.normalclient.get(url)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            json.loads(response.content)["result"]["Facility"], "test facility code"
+        )
+
+    @responses.activate
+    @override_settings(JEMBI_BASE_URL="http://jembi/ws/rest/v1/")
+    def test_facility_code_check_fail(self):
+
+        """
+            Test on Facility Code Check
+            GET - returns 400 response if facility code is incorrect
+        """
+
+        clinic_code = 111111
+
+        self.make_source_normaluser()
+        result = {"request_error": "HTTP 400 Bad Request"}
+        responses.add(
+            responses.GET,
+            "http://jembi/ws/rest/v1/facilityCheck?{}".format(
+                urlencode({"criteria": "value:{}".format(clinic_code)})
+            ),
+            json=result,
+            status=400,
+        )
+        url = "{}?{}".format(
+            reverse("facilitycode-check"), urlencode({"clinic_code": clinic_code})
+        )
+        response = self.normalclient.get(url)
+        self.assertEqual(response.status_code, 400)
+
+    @responses.activate
+    @override_settings(JEMBI_BASE_URL="http://jembi/ws/rest/v1/")
+    def test_facility_code_check_no_code_returned(self):
+
+        """
+            Test on Facility Code Check when Jembi return empty array for
+            wrong code given
+            GET - returns 404 response
+        """
+
+        clinic_code = 111111
+
+        self.make_source_normaluser()
+        result = {"title": "", "headers": [], "rows": [], "width": 0, "height": 0}
+        responses.add(
+            responses.GET,
+            "http://jembi/ws/rest/v1/facilityCheck?{}".format(
+                urlencode({"criteria": "value:{}".format(clinic_code)})
+            ),
+            json=result,
+            status=200,
+        )
+        url = "{}?{}".format(
+            reverse("facilitycode-check"), urlencode({"clinic_code": clinic_code})
+        )
+        response = self.normalclient.get(url)
+        self.assertEqual(response.status_code, 404)
```

Comment:
I'm not sure that we should return a 404 here. If we do, then this goes to the "error" branch of the rapidpro flow, and then it's not easy to know whether the http request failed, or if the user typed in an invalid clinic code.

I think we should instead return a 200, and have the body describe that no clinic was found, either by having a field that's true/false, or by just returning a special value for the name if a clinic isn't found.

Reply:
could I return something like `resp = {"result": {"Facility": "incorrect clinic code"}`?

Reply:
Yup, that would be the "returning a special value" solution. We should probably just make it something like "invalid", I'm not sure how difficult it is to match on a thing with spaces in RapidPro.

Reply:
will do